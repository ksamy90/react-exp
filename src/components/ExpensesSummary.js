import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = ({ expenseCount, amount }) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    const expensesTotal = numeral(amount / 100).format('$0,0.00');
    if (expenseCount === 0) {
        return <p></p>;
    };

    return (
        <div>
            <h1>Viewing {expenseCount} {expenseWord} totalling {expensesTotal}</h1>
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount: visibleExpenses.length,
        amount: selectExpensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);
