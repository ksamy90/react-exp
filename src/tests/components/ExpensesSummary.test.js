import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';
import selectExpensesTotal from '../../selectors/expenses-total';

test('should render ExpensesSummary correctly', () => {
    const wrapper = shallow(<ExpensesSummary />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with expenses', () => {
    const wrapper = shallow(<ExpensesSummary 
        expenseCount={expenses.length} 
        amount={selectExpensesTotal(expenses)}
    />);
    expect(wrapper).toMatchSnapshot();
});
