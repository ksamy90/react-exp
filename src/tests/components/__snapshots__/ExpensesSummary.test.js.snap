// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly 1`] = `
<div>
  <h1>
    Viewing 
     
    expenses
     totalling 
    $0.00
  </h1>
</div>
`;

exports[`should render ExpensesSummary with expenses 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling 
    $1,141.95
  </h1>
</div>
`;
